uniffi_bindgen/src/interface/attributes.rs

```diff
@@ -190,24 +196,29 @@ impl<T: TryInto<FunctionAttributes, Error = anyhow::Error>> TryFrom<Option<T>>
 
 /// Represents UDL attributes that might appear on a function argument.
 ///
-/// This supports the `[ByRef]` attribute for arguments that should be passed
-/// by reference in the generated Rust scaffolding.
+/// This supports the `[ByRef]` and `[ByArc]` attribute for arguments that
+/// should be passed by reference in the generated Rust scaffolding.
 #[derive(Debug, Clone, Hash, Default)]
 pub(super) struct ArgumentAttributes(Vec<Attribute>);
 
 impl ArgumentAttributes {
     pub fn by_ref(&self) -> bool {
         self.0.iter().any(|attr| matches!(attr, Attribute::ByRef))
     }
+    pub fn by_arc(&self) -> bool {
+        self.0.iter().any(|attr| matches!(attr, Attribute::ByArc))
+    }
 }
 
 impl TryFrom<&weedle::attribute::ExtendedAttributeList<'_>> for ArgumentAttributes {
     type Error = anyhow::Error;
     fn try_from(
         weedle_attributes: &weedle::attribute::ExtendedAttributeList<'_>,
     ) -> Result<Self, Self::Error> {
+        // XXX - how do we ensure both `ByRef` and `ByArc` aren't specified?
```

Comment:
FWIW, the `InterfaceAttributes` struct has an extremely dumb check to ensure that `[Threadsafe]` and `[Enum]` are not applied to the same interface.